packages/dx-react-demos/src/bootstrap3/featured-controlled-mode/demo.jsx

```diff
@@ -62,103 +56,103 @@ CommandButton.defaultProps = {
+    text: 'New',
+    hint: 'Create new row',
+  },
+  edit: {
+    text: 'Edit',
+    hint: 'Edit row',
+  },
+  delete: {
+    icon: 'trash',
+    hint: 'Delete row',
+    isDanger: true,
+  },
+  commit: {
+    text: 'Save',
+    hint: 'Save changes',
+  },
+  cancel: {
+    icon: 'remove',
+    hint: 'Cancel changes',
+    isDanger: true,
+  },
 };
 
-const DeleteButton = ({ onExecute }) => (
+const Command = ({ id, onExecute }) => (
   <CommandButton
-    icon="trash"
-    hint="Delete row"
+    {...commandComponentProps[id]}
     onExecute={onExecute}
-    isDanger
   />
 );
-DeleteButton.propTypes = {
+Command.propTypes = {
+  id: PropTypes.string.isRequired,
   onExecute: PropTypes.func.isRequired,
 };
 
-const CommitButton = ({ onExecute }) => (
-  <CommandButton
-    text="Save"
-    hint="Save changes"
-    onExecute={onExecute}
-  />
-);
-CommitButton.propTypes = {
-  onExecute: PropTypes.func.isRequired,
-};
 
-const CancelButton = ({ onExecute }) => (
-  <CommandButton
-    icon="remove"
-    hint="Cancel changes"
-    onExecute={onExecute}
-    isDanger
-  />
-);
-CancelButton.propTypes = {
-  onExecute: PropTypes.func.isRequired,
-};
-
-const commandComponents = {
-  add: AddButton,
-  edit: EditButton,
-  delete: DeleteButton,
-  commit: CommitButton,
-  cancel: CancelButton,
+const availableValues = {
+  product: globalSalesValues.product,
+  region: globalSalesValues.region,
+  customer: globalSalesValues.customer,
 };
 
 export const LookupEditCell = ({
-  column, value, onValueChange,
-}) => {
-  const availableColumnValues = availableValues[column.name];
-  return (
-    <td
-      style={{
-        verticalAlign: 'middle',
-        padding: 1,
-      }}
+  column, availableColumnValues, value, onValueChange,
+}) => (
+  <td
+    style={{
+      verticalAlign: 'middle',
+      padding: 1,
+    }}
+  >
+    <select
+      className="form-control"
+      style={{ width: '100%', textAlign: column.align }}
+      value={value}
+      onChange={e => onValueChange(e.target.value)}
     >
-      <select
-        className="form-control"
-        style={{ width: '100%', textAlign: column.align }}
-        value={value}
-        onChange={e => onValueChange(e.target.value)}
-      >
-        {availableColumnValues.map(val => <option key={val} value={val}>{val}</option>)}
-      </select>
-    </td>
-  );
-};
+      {availableColumnValues.map(val => <option key={val} value={val}>{val}</option>)}
+    </select>
+  </td>
+);
 LookupEditCell.propTypes = {
   column: PropTypes.object.isRequired,
+  availableColumnValues: PropTypes.array.isRequired,
   value: PropTypes.any,
   onValueChange: PropTypes.func.isRequired,
 };
 LookupEditCell.defaultProps = {
   value: undefined,
 };
 
+const Cell = (props) => {
+  if (props.column.name === 'discount') {
+    return <ProgressBarCell {...props} />;
+  }
+  if (props.column.name === 'amount') {
+    return <HighlightedCell {...props} />;
+  }
+  return <Table.Cell {...props} />;
```

Comment:
Can we return `undefined` here?

Reply:
No, a React component can't return `undefined`.